bamToFastq.wdl

```diff
@@ -0,0 +1,341 @@
+        tags = list(set(re.findall(r"[^\t:]+(?=:)", rg)))
+
+        if all(i in tags for i in inputs) == False:  
+            print("FAILED: Missing input tag")
+            d = difflib.Differ()
+            diff = d.compare(tags, inputs)
+            errorMessage = '\n'.join(diff)
+            validate("false")
+            raise ValueError("Input tag does not exist:" + '\n' + errorMessage)
+
+            # Check No. 2:
+            # Predicting modified file name:
+            # Exits if non-unique names are created
+
+        rgArray = []
+        dictArray = []
+
+        for line in rg.splitlines():
+            data = line[1:].split()
+            rgArray.append(data)
+
+        for row in rgArray:
+            del row[0]
+            for i in range(len(row)):
+                k = row[i].split(":")
+                row[i] = k
+            dictArray.append({row[i][0]: row[i][1] for i in range(len(row))})
+
+        fastqNames = []
+        rgData = {}
+
+        for j in range(len(dictArray)):
+            idData = []
+            for readNum in [1, 2]:
+                newName = fileName.format_map(dictArray[j])
+                predictedFileName = f"{newName}_{readNum}.fastq.gz"
+                if predictedFileName in fastqNames:
+                    validate("false")
+                    raise ValueError("File name results in non-unique names")
+                else:
+                    fastqNames.append(predictedFileName)
+                    idData.append(predictedFileName)
+            rgData[dictArray[j]["ID"]]=idData
+
+        print(rgData)
+        
+        CODE
+
+        >>>
+
+        runtime {
+            modules: "~{modules}"
+            memory: "~{memory}G"
+            timeout: "~{timeout}"
+        }
+
+        output {
+            String rgData = read_string(stdout())
+            Boolean valid = read_boolean("isValid.txt")
+
+        }
+
+        meta {
+            output_meta: {
+                rgData: "A Python dictionary containing RG data",
+                valid: "A boolean value to determine workflow continuation"
+            }
+        }
+}
+
+task backExtract {
+        input {
+            File bamFile
+
+            String modules = "picard/2.21.2"
+            Int memory = 24
+            Int timeout = 12
+        }
+
+        parameter_meta {
+            bamFile: "A BAM file with one or more readgroups"
+            modules: "Required environment modules"
+            memory: "Memory allocated for this job"
+            timeout: "Time in hours before task timeout"
+        }
+
+        command <<<
+            module unload cromwell ## <- only needed for local testing
+            module unload java ## <- only needed for local testing
+            module load picard
+
+            java -Xmx20g -jar $PICARD_ROOT/picard.jar SamToFastq \
+            INPUT=~{bamFile} \
+            RG_TAG="ID" \
+            OUTPUT_DIR=. \
+            OUTPUT_PER_RG=true \
+            NON_PF=true \
```

Comment:
I would consider exposing at least some of these variables, perhaps NON_PF and OUTPUT_PER_RG? You may always set them to pre-selected default values